src/IO/Importers/ElementActions.hpp

```diff
@@ -53,5 +61,111 @@ struct RegisterWithVolumeDataReader {
   }
 };
 
-}  // namespace Actions
-}  // namespace importers
+/*!
+ * \brief Read a volume data file and distribute the data to all registered
+ * elements.
+ *
+ * Invoke this action on the elements of an array parallel component to dispatch
+ * reading the volume data file specified by the options in
+ * `ImporterOptionsGroup`. The tensors in `FieldTagsList` will be loaded from
+ * the file and distributed to all elements that have previously registered. Use
+ * `importers::Actions::RegisterWithVolumeDataReader` to register the elements
+ * of the array parallel component in a previous phase.
+ *
+ * Note that the volume data file will only be read once per node, triggered by
+ * the first element that invokes this action. All subsequent invocations of
+ * this action on the node will do nothing. See
+ * `importers::Actions::ReadAllVolumeDataAndDistribute` for details.
+ *
+ * The data is distributed to the elements using `Parallel::receive_data`. The
+ * elements can monitor `importers::Tags::VolumeData` in their inbox to wait for
+ * the data and process it once it's available. We provide the action
+ * `importers::Actions::ReceiveVolumeData` that waits for the data and moves it
+ * directly into the DataBox. You can also implement a specialized action that
+ * might verify and post-process the data before populating the DataBox.
+ *
+ * \see Dev guide on \ref dev_guide_importing
+ */
+template <typename ImporterOptionsGroup, typename FieldTagsList>
+struct ReadVolumeData {
```

Comment:
why does this action exist?  why can't the `VolumeDataReader` invoke `ReadAllVolumeDataAndDistribute` in its own phase action list?

Reply:
Synchronizing PDALs between parallel components doesn't work very well. We could construct the PDALs for the `ElementDataReader` nodegroup in the metavariables, including the `InitializeElementDataReader` action for the init phase. Not sure if that's preferable.

Reply:
okay, thank you for looking into this; it is not worth holding up this PR